dvc/remote/gdrive/__init__.py

```diff
@@ -0,0 +1,207 @@
+from __future__ import unicode_literals
+
+import os
+
+from funcy import cached_property
+import ratelimit
+
+from dvc.scheme import Schemes
+from dvc.path_info import CloudURLInfo
+from dvc.remote.base import RemoteBASE
+from dvc.config import Config
+from dvc.remote.gdrive.utils import TrackFileReadProgress, shared_token_warning
+
+
+class GDriveURLInfo(CloudURLInfo):
+    @property
+    def netloc(self):
+        return self.parsed.netloc
+
+
+class RemoteGDrive(RemoteBASE):
+    scheme = Schemes.GDRIVE
+    path_cls = GDriveURLInfo
+    REGEX = r"^gdrive://.*$"
+    REQUIRES = {"pydrive": "pydrive"}
+    PARAM_CHECKSUM = "md5Checksum"
+    DEFAULT_GOOGLE_AUTH_SETTINGS_PATH = os.path.join(
+        os.path.dirname(__file__), "settings.yaml"
+    )
+    FOLDER_MIME_TYPE = "application/vnd.google-apps.folder"
+
+    def __init__(self, repo, config):
+        super(RemoteGDrive, self).__init__(repo, config)
+        if Config.SECTION_GDRIVE_CREDENTIALPATH not in config:
+            shared_token_warning()
```

Comment:
I wonder if we should actually not provide shared token at all. We will run out of quotas extremely quick and it would look like dvc is broken.

Reply:
@efiop do you mean here to not provide default `client_id` and `client_secret`? The default max quota for requests in the worst scenario seems is just 10 simultaneous users syncing huge repos.
Does it makes sense to ask user interactively to enter `client_id` and `client_server` after they will be generated by user (it might be possible to put link to the on-line docs with details how to create required credentials)?

Reply:
@MaxRis I know that we've discussed this a lot already, but to refresh the context - what are the default quotas for rclone? And how do they solve the problem?

Am I right that the decision was to exclude default credentials, right? Or do we still keep them?

Reply:
> what are the default quotas for rclone? And how do they solve the problem?

rclone has doc for Google Drive - https://rclone.org/drive/
they seems use the highest possible for them quotas ( https://rclone.org/drive/#making-your-own-client-id ):
```
When you use rclone with Google drive in its default configuration you are using rclone’s client_id. This is shared between all the rclone users. There is a global rate limit on the number of queries per second that each client_id can do set by Google. rclone already has a high quota and I will continue to make sure it is high enough by contacting Google.

It is strongly recommended to use your own client ID as the default rclone ID is heavily used.
```

Also rclone has `an interactive setup process` for Google Drive https://rclone.org/drive/#i-class-fab-fa-google-i-google-drive

> Am I right that the decision was to exclude default credentials, right? Or do we still keep them?

Yes, DVC errors out if custom `gdrive_client_id` and `gdrive_client_secret` are not set in the config.

Reply:
What does it take to get Google approve higher limits?

Reply:
User should enter name/email/contact phone and the new value for quotas in special request form(click at [EDIT QUOTAS](https://console.cloud.google.com/iam-admin/quotas)) for the specific Google Project.
![request-higher-GDrive-quotas](https://user-images.githubusercontent.com/1928902/69756268-e6f12880-110e-11ea-8ee4-f9a718af4bfd.png)

Reply:
Can we apply now?

Reply:
As discussed further, it's supposed to have Iterative Google Project with the highest quotas. Requirements are added to the To-Do list as part of issue https://github.com/iterative/dvc/issues/2865
Resolving this for now.